src/tools/perf/lib/libperf.c

```diff
@@ -1576,12 +1606,7 @@ void uct_perf_barrier(ucx_perf_context_t *perf)
 
 void ucp_perf_barrier(ucx_perf_context_t *perf)
 {
-    rte_call(perf, barrier, (void(*)(void*))ucp_worker_progress,
-#if _OPENMP
-             (void*)perf->ucp.tctx[omp_get_thread_num()].perf.ucp.worker);
-#else
-             (void*)perf->ucp.tctx[0].perf.ucp.worker);
-#endif
+    rte_call(perf, barrier, ucp_perf_worker_progress, perf);
```

Comment:
need to make sure that ucp_worker is not created with SINGLE thread mode (only SERIALIZED)

Reply:
why needed? the barrier is called from a master thread only

Reply:
called from threads in : https://github.com/openucx/ucx/blob/dad4ad2b520323b71915f16306fe51b71aafec85/src/tools/perf/lib/libperf_thread.c#L52

Reply:
done